src/Autofac/Core/Resolving/ActivatorExtensions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// This software is part of the Autofac IoC container
+// Copyright © 2019 Autofac Contributors
+// https://autofac.org
+//
+// Permission is hereby granted, free of charge, to any person
+// obtaining a copy of this software and associated documentation
+// files (the "Software"), to deal in the Software without
+// restriction, including without limitation the rights to use,
+// copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following
+// conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+// OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+// WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+// OTHER DEALINGS IN THE SOFTWARE.
+
+using Autofac.Core.Activators.Delegate;
+
+namespace Autofac.Core.Resolving
+{
+    internal static class ActivatorExtensions
+    {
+        // This shorthand name for the activator is used in exception messages; for activator types
+        // where the limit type generally describes the activator exactly, we use that; for delegate
+        // activators, a variation on the type name is used to indicate this.
+        public static string DisplayName(this IInstanceActivator activator)
+        {
+            var fullName = activator.LimitType.FullName ?? "";
+            return activator is DelegateActivator ?
+                $"λ:{fullName}" :
```

Comment:
Would it be interesting to have info about whether it was reflection or provided instance activation?

Reply:
Possibly; I didn't add this because there are few (no?) scenarios where a provided instance will trigger an activation exception.

I also was aiming to describe the type of the component, rather than its internal representation in Autofac - delegates are the special case because their `LimitType` might be surprising, e.g. `System.Object` when the actual component is one of several other alternatives.